client/network-gossip/src/state_machine.rs

```diff
@@ -572,4 +590,52 @@ mod tests {
 		consensus.peer_disconnected(&mut network, peer_id.clone());
 		assert!(!consensus.peers.contains_key(&peer_id));
 	}
+
+	#[test]
+	fn on_incoming_ignores_discarded_messages() {
+		let to_forward = ConsensusGossip::<Block>::new(
+			Arc::new(DiscardAll),
+			[0, 0, 0, 0],
+		).on_incoming(
+			&mut NoOpNetwork::default(),
+			PeerId::random(),
+			vec![vec![1, 2, 3]],
+		);
+
+		assert!(
+			to_forward.is_empty(),
+			"Expected `on_incoming` to ignore discarded message but got {:?}", to_forward,
+		);
+	}
+
+	#[test]
+	fn on_incoming_ignores_unregistered_peer() {
+		let mut network = NoOpNetwork::default();
+		let remote = PeerId::random();
+
+		let to_forward = ConsensusGossip::<Block>::new(
+			Arc::new(AllowAll),
+			[0, 0, 0, 0],
+		).on_incoming(
+			&mut network,
+			// Unregistered peer.
+			remote.clone(),
+			vec![vec![1, 2, 3]],
+		);
+
+		assert!(
+			to_forward.is_empty(),
+			"Expected `on_incoming` to ignore message from unregistered peer but got {:?}",
+			to_forward,
+		);
+
+		assert_eq!(
+			vec![
+				(remote.clone(), rep::GOSSIP_SUCCESS),
```

Comment:
See comment above. This replicates the behaviour of the previous implementation.